Realm/ObjectServerTests/SwiftObjectServerTests.swift

```diff
@@ -521,27 +669,123 @@ class SwiftObjectServerTests: SwiftSyncTestCase {
+                XCTAssertEqual(realm.objects(SwiftPerson.self)[0].firstName, "John")
+            }
+            resetSyncManager()
+        }
+
+        let user = try logInUser(for: creds)
         var configuration = user.configuration(partitionValue: #function)
         configuration.objectTypes = [SwiftPerson.self]
 
-        let ex = self.expectation(description: "get client reset error")
-        let syncManager = self.app.syncManager
-        syncManager.errorHandler = { error, session in
-            guard let error = error as? SyncError else {
-                XCTFail("Bad error type: \(error)")
-                return
+        try autoreleasepool {
+            let realm = try Realm(configuration: configuration)
+            waitForDownloads(for: realm)
+            // After reopening, the old Realm file should have been moved aside
+            // and we should now have the data from the server
+            XCTAssertEqual(realm.objects(SwiftPerson.self).count, 1)
+            XCTAssertEqual(realm.objects(SwiftPerson.self)[0].firstName, "Paul")
+        }
+    }
+
+    func testClientResetManualNoCallbacks() throws {
+        let creds = basicCredentials()
+        try autoreleasepool {
+            let user = try logInUser(for: creds)
+            try prepareClientReset(#function, user)
+
+            // No callback is passed into enum `.manual`, and no syncManager.errorHandler exists
+            var configuration = user.configuration(partitionValue: #function, clientResetMode: .manual())
+            configuration.objectTypes = [SwiftPerson.self]
+
+            try autoreleasepool {
+                let realm = try Realm(configuration: configuration)
+                // The locally created object should still be present as we didn't
+                // actually handle the client reset
+                XCTAssertEqual(realm.objects(SwiftPerson.self).count, 1)
+                XCTAssertEqual(realm.objects(SwiftPerson.self)[0].firstName, "John")
             }
-            XCTAssertEqual(error.code, .clientResetError)
-            XCTAssertEqual(session?.state, .inactive)
-            XCTAssertEqual(session?.connectionState, .disconnected)
-            XCTAssertEqual(session?.parentUser()?.id, user.id)
-            ex.fulfill()
+            resetSyncManager()
+        }
+
+        let user = try logInUser(for: creds)
+        var configuration = user.configuration(partitionValue: #function)
+        configuration.objectTypes = [SwiftPerson.self]
+
+        try autoreleasepool {
+            let realm = try Realm(configuration: configuration)
+            waitForDownloads(for: realm)
+            // After reopening, the old Realm file should have been moved aside
+            // and we should now have the data from the server
+            XCTAssertEqual(realm.objects(SwiftPerson.self).count, 1)
+            XCTAssertEqual(realm.objects(SwiftPerson.self)[0].firstName, "Paul")
         }
+    }
```

Comment:
We may need an extra test that makes sure the manual mode callback is called even if the error handler is set as well